gc/base/standard/Scavenger.cpp

```diff
@@ -4688,28 +4696,11 @@ MM_Scavenger::threadFinalReleaseCopyCaches(MM_EnvironmentBase *envBase, MM_Envir
 		 * In a case of thread teardown or flushing caches for walk, caller ensures that own environment is used.
 		 */
 
-		// todo: try to simplify. perhaps even we can use threadEnvironment directly?
+		abandonSurvivorTLHRemainder(threadEnvironment);
+		abandonTenureTLHRemainder(threadEnvironment, true);
 
 		/* take a snapshot of all copy cache relevant variables */
 		if (envBase != threadEnvironmentBase) {
```

Comment:
I belive we can remove this if condition. the asserts should hold true for any thread. you'll need to retest (z/os 64bit only) after the change